common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -506,6 +506,12 @@ def editor_saved(self, user, old_metadata, old_content):
 
         if metadata_was_changed_by_user:
             self.edx_video_id = self.edx_video_id.strip()
+            # override `youtube_id_1_0` if there is a val profile for youtube.
+            if self.edx_video_id and edxval_api:
+                val_youtube_id = edxval_api.get_url_for_profile(self.edx_video_id, "youtube")
+                if val_youtube_id:
+                    self.youtube_id_1_0 = val_youtube_id
```

Comment:
This will also show the latest transcript after we save the component for first time.